### DISCUSSION TITLE
Proper way to close remaining windows

### DISCUSSION THREAD

**Discussion by egebilecen** [tauri-apps/tauri#8503] [Q&A] [answered]
Hello everyone. Currently my application creates couple webview windows. But when I close the main window, these webview windows are still left open and main executable still runs until these windows are also closed. My workaround was to exit the application from handle when the main window is closed but when I do this, following error pops-up to the console: `[1230/145847.989:ERROR:window_impl.cc(120)] Failed to unregister class Chrome_WidgetWin_0. Error = 0`. Is there a better way of doing this? Thank you.

My code for achieving this is the following at the moment:
```rust
fn main() {
    let tauri_builder = tauri::Builder::default().invoke_handler(tauri::generate_handler![...]);

    ...

    tauri_builder
        .on_window_event(|event| match event.event() {
            tauri::WindowEvent::Destroyed => {
                let window = event.window();
                if window.label() == "main" {
                    window.app_handle().exit(0);
                }
            }
            _ => {}
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

**Comment by FabianLars** [comment, marked as answer]
You can get all windows via `window.app_handle().windows()` and close them individually, _but_ that may also trigger this warning since it's something chrome internal. You can even often trigger that same warning in a normal browser so i wouldn't worry about it.

**Reply by egebilecen** [reply]
Thank you, then I will just stick with current approach!
